projects/rero/ng-core/src/lib/record/editor/formly/primeng/input/src/input.type.ts

```diff
@@ -49,15 +46,33 @@ export interface NgCoreFormlyInputFieldConfig extends FormlyFieldConfig {
           }
       </p-inputgroup>
     } @else {
-      <input
-        pInputText
-        [class]="props.class"
-        [type]="props.type || 'text'"
-        [formControl]="formControl"
-        [formlyAttributes]="field"
-        [ngClass]="{ 'ng-invalid ng-dirty': showError }"
+      <ng-container
+        [ngTemplateOutlet]="input"
+        [ngTemplateOutletContext]="{ formControl, props, field, showError }"
       />
     }
+    <ng-template #input let-formControl="formControl" let-props="props" let-field="field" let-showError="showError">
+      @if (props.type === 'number') {
+        <input
+          pInputText
+          [class]="props.class"
+          type="number"
+          [attr.step]="props.inputStep"
+          [formControl]="formControl"
+          [formlyAttributes]="field"
+          [ngClass]="{ 'ng-invalid ng-dirty': showError }"
+        />
+      } @else {
+        <input
+          pInputText
+          [class]="props.class"
+          [type]="props.type || 'text'"
```

Comment:
you can try `[attr.type]` instead.

Reply:
it doesn't work with `[attr.type]` either